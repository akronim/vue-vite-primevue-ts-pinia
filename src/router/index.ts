import { createRouter, createWebHistory, type RouteComponent, type RouteRecordRaw } from 'vue-router'

export const RouteNames = {
  VUE_TEST_ONE: `vueTestOne`,
  VUE_TEST_TWO: `vueTestTwo`
}

export const routes: RouteRecordRaw[] = [
  {
    path: `/demo/vuetest/one`,
    name: RouteNames.VUE_TEST_ONE,
    component: (): Promise<RouteComponent> => import(`@/views/demo/VueTestOne.vue`)
  },
  {
    path: `/demo/vuetest/two`,
    name: RouteNames.VUE_TEST_TWO,
    component: (): Promise<RouteComponent> => import(`@/views/demo/VueTestTwo.vue`)
  }
]

export const router = createRouter({
  routes,
  //history: createWebHistory(import.meta.env.BASE_URL),
  history: createWebHistory()
})