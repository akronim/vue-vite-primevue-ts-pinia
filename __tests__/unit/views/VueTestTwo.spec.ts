import { describe, it, expect, beforeEach } from 'vitest'
import { VueWrapper, mount } from '@vue/test-utils'
import VueTestTwo from '@/views/demo/VueTestTwo.vue'
import { createRouter, createWebHistory } from 'vue-router'
import { routes } from '@/router'
import { h } from 'vue'

const mockRouter = createRouter({
  history: createWebHistory(),
  routes: [
    { path: `/`, component: h(`div`) },
    ...routes
  ]
})

describe(`VueTestTwo`, () => {
  const options = {
    global: {
      plugins: [mockRouter]
    }
  }

  let wrapper: VueWrapper<any>

  beforeEach(async () => {
    wrapper = mount(VueTestTwo, options)
  })

  it(`renders properly`, async () => {
    expect(wrapper.isVisible()).toBeTruthy()
  })

  it('snapshot UI testing', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })
})

