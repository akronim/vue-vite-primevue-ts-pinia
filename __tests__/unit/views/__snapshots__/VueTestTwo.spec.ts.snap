// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueTestTwo > snapshot UI testing 1`] = `"Go To OneCount0"`;
